frontend/src/components/common/Pagination.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useState } from "react";
+import {
+  Menu,
+  MenuButton,
+  MenuList,
+  MenuItem,
+  Button,
+  NumberInput,
+  NumberInputField,
+  IconButton,
+  Flex,
+  Box,
+  Text,
+  GridItem,
+} from "@chakra-ui/react";
+
+import {
+  ChevronLeftIcon,
+  ChevronRightIcon,
+  ChevronDownIcon,
+  TriangleDownIcon,
+} from "@chakra-ui/icons";
+import { getLocalStorageObjProperty } from "../../utils/LocalStorageUtils";
+
+type Props = {
+  numRecords: number;
+  resultsPerPage: number;
+  pageNum: number;
+  userPageNum: number;
+  setUserPageNum: React.Dispatch<React.SetStateAction<number>>;
+  setResultsPerPage: React.Dispatch<React.SetStateAction<number>>;
+  getLogRecords: (page_number: number) => Promise<void>;
+};
+
+const RESULTS_PER_PAGE_OPTIONS = [10, 25, 50];
+
+const Pagination = ({
+  numRecords,
+  resultsPerPage,
+  pageNum,
+  userPageNum,
+  setUserPageNum,
+  setResultsPerPage,
+  getLogRecords,
+}: Props): React.ReactElement => {
+  const numPages = Math.ceil(numRecords / resultsPerPage);
+
+  const handleNumberInputChange = (
+    newUserPageNumString: string,
+    newUserPageNum: number,
+  ) => {
+    setUserPageNum(newUserPageNum);
+    if (
+      newUserPageNum !== pageNum &&
+      newUserPageNum >= 1 &&
+      newUserPageNum <= numPages
+    ) {
+      getLogRecords(newUserPageNum);
+    }
+  };
+
+  const handleBlur = (event: React.FocusEvent<HTMLInputElement>) => {
+    if (Number.isNaN(userPageNum)) {
+      setUserPageNum(pageNum);
+    }
+  };
+
+  const handlePageArrowPress = (newUserPageNum: number) => {
+    setUserPageNum(newUserPageNum);
+    getLogRecords(newUserPageNum);
+  };
+
+  const getNumRecordsStr = () => {
+    return `${numRecords} ${numRecords === 1 ? "item" : "items"}`;
+  };
+
+  return (
+    <Box padding="12px 0px 33px">
+      <Flex justifyContent="space-between" alignItems="center">
+        <Box>
+          <Text color="#6D8788">{getNumRecordsStr()}</Text>
+        </Box>
+        <Box>
+          <Flex alignItems="center" justifyContent="space-evenly">
+            <IconButton
+              className="icon-button"
+              variant="ghost"
+              aria-label="Previous page"
+              icon={<ChevronLeftIcon boxSize={7} color="#6D8788" />}
+              disabled={userPageNum <= 1}
+              onClick={() => handlePageArrowPress(userPageNum - 1)}
+            />
+            <Flex
+              alignItems="center"
+              justifyContent="space-evenly"
+              padding="0px 25px 0px"
+              width="210px"
+            >
+              <Text>Page</Text>
+              <NumberInput
+                maxW="60px"
```

Comment:
an issue with this is if there are 1000s of pages, the input box only shows 2 digits. This may be confusing to the user. I don't know what the width should be, but my (naïve) solution would be just to use a width that accommodates the largest possible integer. 

Syncing with the designers on this would be a good first step.